sdk/src/Services/DynamoDBv2/Custom/DataModel/Context.cs

```diff
@@ -401,23 +419,43 @@ private async Task SaveHelperAsync([DynamicallyAccessedMembers(InternalConstants
             if (storage == null) return;
 
             Table table = GetTargetTable(storage.Config, flatConfig);
+
+            var counterConditionExpression = BuildCounterConditionExpression(storage);
+
+            Document updateDocument;
+            Expression versionExpression = null;
             if (
                 (flatConfig.SkipVersionCheck.HasValue && flatConfig.SkipVersionCheck.Value)
                 || !storage.Config.HasVersion)
             {
-                await table.UpdateHelperAsync(storage.Document, table.MakeKey(storage.Document), null, cancellationToken).ConfigureAwait(false);
+                updateDocument = await table.UpdateHelperAsync(storage.Document, table.MakeKey(storage.Document), new UpdateItemOperationConfig
+                {
+                    ReturnValues = ReturnValues.AllNewAttributes
```

Comment:
To avoid change in behavior should we set `AllNewAttributes` only if there is a counter condition expression?